This patch is based on the commit described below
modified by Peter Michael Green to apply to
rust-rand 0.7.3

commit fe34550f4ef28056c73a785f54f908f8fc13070b
Author: Vinzent Steinberg <Vinzent.Steinberg@gmail.com>
Date:   Thu Oct 8 18:36:49 2020 +0200

    Remove `__m64` support
    
    This is necessary, because support for `__m64` was removed from nighly
    Rust [1].
    
    Fixes #1050.
    
    [1] https://github.com/rust-lang/stdarch/issues/823

Index: rand/src/distributions/integer.rs
===================================================================
--- rand.orig/src/distributions/integer.rs
+++ rand/src/distributions/integer.rs
@@ -158,7 +158,7 @@ simd_impl!(512, u8x64, i8x64, u16x32, i1
     feature = "nightly",
     any(target_arch = "x86", target_arch = "x86_64")
 ))]
-simd_impl!((__m64, u8x8), (__m128i, u8x16), (__m256i, u8x32),);
+simd_impl!((__m128i, u8x16), (__m256i, u8x32),);
 
 #[cfg(test)]
 mod tests {
Index: rand/src/distributions/utils.rs
===================================================================
--- rand.orig/src/distributions/utils.rs
+++ rand/src/distributions/utils.rs
@@ -161,7 +161,6 @@ mod simd_wmul {
     }
 
     wmul_impl! { (u16x2, u32x2),, 16 }
-    #[cfg(not(target_feature = "sse2"))]
     wmul_impl! { (u16x4, u32x4),, 16 }
     #[cfg(not(target_feature = "sse4.2"))]
     wmul_impl! { (u16x8, u32x8),, 16 }
@@ -188,8 +187,6 @@ mod simd_wmul {
         };
     }
 
-    #[cfg(target_feature = "sse2")]
-    wmul_impl_16! { u16x4, __m64, _mm_mulhi_pu16, _mm_mullo_pi16 }
     #[cfg(target_feature = "sse4.2")]
     wmul_impl_16! { u16x8, __m128i, _mm_mulhi_epu16, _mm_mullo_epi16 }
     #[cfg(target_feature = "avx2")]
